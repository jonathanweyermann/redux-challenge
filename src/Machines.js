import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Health from './health';
import './machines.css';
import { Table } from 'react-bootstrap';
import { getMachines } from '../src/actions';

function Machines (props) {

	useEffect(() => {
		props.getMachines();
	}, []);

	const machineDetails = (id) => {
		props.history.push(`/machines/${id}`);
	};

	const machinesLayout = () => {
		let layout = props.machines.machines.map((machine) => {
			return (
				<tr key={machine.id} onClick={() => machineDetails(machine.id)}>
					<td>{machine.name}</td>
					<td colSpan="2">{machine.ip_address}</td>
					<td colSpan="2"><Health value={machine.health} /></td>
				</tr>
			);
		});
		return (<React.Fragment>{layout}</React.Fragment>);
	};


	return (
		<Table striped>
			<thead>
				<tr>
					<th>Name</th>
					<th colSpan="2">IP Address</th>
					<th colSpan="2">Health</th>
				</tr>
			</thead>
			<tbody>
				{ machinesLayout() }
			</tbody>
		</Table>
	);
}

const mapStateToProps = function (state) {
	return {
		machines: state.machines
	};
};

Machines.propTypes = {
	getMachines: PropTypes.func,
	machines: PropTypes.object,
	history: PropTypes.object
};


export default connect(mapStateToProps, { getMachines })(Machines);
