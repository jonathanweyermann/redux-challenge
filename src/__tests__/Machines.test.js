import { render, fireEvent } from '@testing-library/react';
import { expect as chai_expect } from 'chai';
import renderer from 'react-test-renderer';
var sinon = require('sinon');

import React from 'react';
import { shallow, mount } from 'enzyme';
import Machines from '../Machines';

import { Provider } from 'react-redux';
import configureMockStore from "redux-mock-store";
import thunk from 'redux-thunk';
const mockStore = configureMockStore([thunk]);
const props = {"history":{"length":5,"action":"POP","location":{"pathname":"/machines","search":"","hash":"","key":"ybxgrt"}}};
const store = mockStore(
  {"machines":{"machines":[{"id":"99ade105-dee1-49eb-8ac4-e4d272f89fba","name":"Machine X","ip_address":"127.0.0.31","health":84},
													 {"id":"4111947a-6c58-4977-90fa-2caaaef88648","name":"Machine 2","ip_address":"127.0.0.4","health":95},
													 {"id":"57342663-909c-4adf-9829-6dd1a3aa9143","name":"Machine 3","ip_address":"127.0.0.55","health":2},
													 {"id":"5632e1ec-46cb-4895-bc8b-a91644568cd5","name":"Machine 4","ip_address":"127.0.0.3","health":22}],
							"machine":{"name":""},
							"name":""}}
);


describe('machines component', () => {
	const setUpFn = () => {
	 return mount(
		 <Provider store={store}>
			 <Machines {...props} />
		 </Provider>
	 );
 };

	let wrapper;
	beforeEach(() => {
		wrapper = setUpFn();
	});

  it('renders 4 machines', () => {
	  expect(wrapper).toMatchSnapshot();
  });

	const setup = () => {
	  const utils = render(<Provider store={store}><Machines {...props} /></Provider>);
	  const input = utils.getByLabelText('name-input')
	  return {
	    input,
	    ...utils,
	  }
	}

	it('renders name of all machines', () => {
	  const { getByText } = render(<Provider store={store}><Machines {...props} /></Provider>);
	  const linkElement = getByText(/Machine X/i);
	  expect(linkElement).toBeInTheDocument();
	  const linkElement2 = getByText(/Machine 2/i);
	  expect(linkElement2).toBeInTheDocument();
	  const linkElement3 = getByText(/Machine 3/i);
	  expect(linkElement3).toBeInTheDocument();
	  const linkElement4 = getByText(/Machine 4/i);
	  expect(linkElement4).toBeInTheDocument();
	});

});
