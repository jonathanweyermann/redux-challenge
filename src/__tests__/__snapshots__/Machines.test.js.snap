// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`machines component renders 4 machines 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Machines)
    history={
      Object {
        "action": "POP",
        "length": 5,
        "location": Object {
          "hash": "",
          "key": "ybxgrt",
          "pathname": "/machines",
          "search": "",
        },
      }
    }
  >
    <Machines
      getMachines={[Function]}
      history={
        Object {
          "action": "POP",
          "length": 5,
          "location": Object {
            "hash": "",
            "key": "ybxgrt",
            "pathname": "/machines",
            "search": "",
          },
        }
      }
      machines={
        Object {
          "machine": Object {
            "name": "",
          },
          "machines": Array [
            Object {
              "health": 84,
              "id": "99ade105-dee1-49eb-8ac4-e4d272f89fba",
              "ip_address": "127.0.0.31",
              "name": "Machine X",
            },
            Object {
              "health": 95,
              "id": "4111947a-6c58-4977-90fa-2caaaef88648",
              "ip_address": "127.0.0.4",
              "name": "Machine 2",
            },
            Object {
              "health": 2,
              "id": "57342663-909c-4adf-9829-6dd1a3aa9143",
              "ip_address": "127.0.0.55",
              "name": "Machine 3",
            },
            Object {
              "health": 22,
              "id": "5632e1ec-46cb-4895-bc8b-a91644568cd5",
              "ip_address": "127.0.0.3",
              "name": "Machine 4",
            },
          ],
          "name": "",
        }
      }
    >
      <ForwardRef
        striped={true}
      >
        <table
          className="table table-striped"
        >
          <thead>
            <tr>
              <th>
                Name
              </th>
              <th
                colSpan="2"
              >
                IP Address
              </th>
              <th
                colSpan="2"
              >
                Health
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              key="99ade105-dee1-49eb-8ac4-e4d272f89fba"
              onClick={[Function]}
            >
              <td>
                Machine X
              </td>
              <td
                colSpan="2"
              >
                127.0.0.31
              </td>
              <td
                colSpan="2"
              >
                <Health
                  value={84}
                >
                  <div
                    className="container"
                    style={
                      Object {
                        "padding": 0,
                      }
                    }
                  >
                    <div
                      className="full-bar"
                    >
                      <div
                        className="solid-bar"
                        data-testid="solid-bar"
                        style={
                          Object {
                            "backgroundColor": "#5cb85c",
                            "width": "calc(100% * 0.84)",
                          }
                        }
                      />
                    </div>
                  </div>
                </Health>
              </td>
            </tr>
            <tr
              key="4111947a-6c58-4977-90fa-2caaaef88648"
              onClick={[Function]}
            >
              <td>
                Machine 2
              </td>
              <td
                colSpan="2"
              >
                127.0.0.4
              </td>
              <td
                colSpan="2"
              >
                <Health
                  value={95}
                >
                  <div
                    className="container"
                    style={
                      Object {
                        "padding": 0,
                      }
                    }
                  >
                    <div
                      className="full-bar"
                    >
                      <div
                        className="solid-bar"
                        data-testid="solid-bar"
                        style={
                          Object {
                            "backgroundColor": "#5cb85c",
                            "width": "calc(100% * 0.95)",
                          }
                        }
                      />
                    </div>
                  </div>
                </Health>
              </td>
            </tr>
            <tr
              key="57342663-909c-4adf-9829-6dd1a3aa9143"
              onClick={[Function]}
            >
              <td>
                Machine 3
              </td>
              <td
                colSpan="2"
              >
                127.0.0.55
              </td>
              <td
                colSpan="2"
              >
                <Health
                  value={2}
                >
                  <div
                    className="container"
                    style={
                      Object {
                        "padding": 0,
                      }
                    }
                  >
                    <div
                      className="full-bar"
                    >
                      <div
                        className="solid-bar"
                        data-testid="solid-bar"
                        style={
                          Object {
                            "backgroundColor": "#D9534F",
                            "width": "calc(100% * 0.02)",
                          }
                        }
                      />
                    </div>
                  </div>
                </Health>
              </td>
            </tr>
            <tr
              key="5632e1ec-46cb-4895-bc8b-a91644568cd5"
              onClick={[Function]}
            >
              <td>
                Machine 4
              </td>
              <td
                colSpan="2"
              >
                127.0.0.3
              </td>
              <td
                colSpan="2"
              >
                <Health
                  value={22}
                >
                  <div
                    className="container"
                    style={
                      Object {
                        "padding": 0,
                      }
                    }
                  >
                    <div
                      className="full-bar"
                    >
                      <div
                        className="solid-bar"
                        data-testid="solid-bar"
                        style={
                          Object {
                            "backgroundColor": "#D9534F",
                            "width": "calc(100% * 0.22)",
                          }
                        }
                      />
                    </div>
                  </div>
                </Health>
              </td>
            </tr>
          </tbody>
        </table>
      </ForwardRef>
    </Machines>
  </Connect(Machines)>
</Provider>
`;
